source/fundamentals/logging.txt

```diff
@@ -0,0 +1,370 @@
+   	ApplyURI(uri).
+   	SetLoggerOptions(loggerOptions)
+
+   client, err := mongo.Connect(context.TODO(), clientOptions)
+
+Configure a Logger
+------------------
+
+To create a ``LoggerOptions`` object, call the ``options.Logger()``
+method. The following table describes how to set properties on
+``LoggerOptions`` to configure your logger. The first column contains
+the ``LoggerOptions`` properties, the second column describes the
+properties, and the third column describes the
+setter methods:
+
+.. list-table::
+   :widths: 31 38 31
+   :header-rows: 1
+
+   * - Property
+     - Description
+     - Setter Method
+
+   * - | ``ComponentLevels``
+       | **Type**: ``map[LogComponent]LogLevel``
+     - | A mapping of components, specified by ``LogComponent``, to log
+         severity levels, specified by ``LogLevel``. The driver uses the
+         ``LogLevel`` for each ``LogComponent`` to determine if the log
+         message should be generated. To learn more about
+         ``LogComponent`` and ``LogLevel``, see :ref:`Logging Components
+         and Severity Levels <golang-logging-component-severity>`.
+     - | ``SetComponentLevel()``
+       | **Parameters**: ``LogComponent``, ``LogLevel``
+
+   * - | ``Sink``
+       | **Type**: ``LogSink``
+     - | The logging interface that the driver uses to log messages.
+         The ``LogSink`` type is an interface that you can implement to
+         provide a :ref:`custom sink <golang-logging-custom>` or integrate a :ref:`third-party
+         logger <golang-logging-thirdparty>` for the driver's logs. If you don't set this
+         property, driver uses the standard logging library.
+     - | ``SetSink()``
+       | **Parameter**: ``LogSink``
+
+   * - | ``MaxDocumentLength``
+       | **Type**: ``uint``
+       | **Default**: ``1000``
+     - | The maximum length in bytes of a document that the driver logs. If the
+         underlying document is larger than this value, the driver
+         truncates it and appends ellipses to the partial log message.
+     - | ``SetMaxDocumentLength()``
+       | **Parameter**: ``uint``
+
+.. tip:: Write Logs to a Specific File
+   
+   By default, the standard logger emits messages to your console (``stderr``). You can
+   specify a logging destination by setting the ``MONGODB_LOG_PATH``
+   environment variable to a filepath or ``stdout``.
+
+.. _golang-logging-component-severity:
+
+Logging Components and Severity Levels
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can specify which components the driver will log against by
+specifying the ``LogComponent`` type. The driver provides the following
+built-in specifications for ``LogComponent``:
+
+- ``LogComponentAll``: enables logging for all components.
+- ``LogComponentCommand``: enables command monitor logging.
+- ``LogComponentTopology``: enables topology logging.
+- ``LogComponentServerSelection``: enables server selection logging.
+- ``LogComponentConnection``: enables connection services logging.
```

Comment:
Q: Would this be easier to read as a table? That has the added benefit of letting you include the corresponding enum values, which you mention in the next paragraph.

Ditto for LogLevel section later.

Reply:
Changed